### PR TITLE
chore: add scalafix

### PR DESCRIPTION
**Pull request by kpodsiad**
We have to agree on scalafix settings. Currently, they are very similar to the ones that can be found in Metals.

What I've found is that `RemoveUnused` have strange behavior - sometimes it removes only `val name =` and lefts right side of the assignment :/ 
Moreover, we have to be very careful, for example here https://github.com/scalacenter/bloop/blob/1b6633ae67ce0ba508caa947cf88b3daf459ee96/frontend/src/main/scala/bloop/Cli.scala#L508-L518 
there is a side effect (`runAsync`) that is assigned to the variable and we can't remove both sides. Probably the best way would be to iterate manually over those unused warnings.

scalafix can be run on frontend by `frontend/scalafix`

cc: @tgodzik

### CODE REVIEW

File: .scalafix.conf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+rules = [ 
+  DisableSyntax,
+  OrganizeImports,
+  NoValInForComprehension,
```
Comment (tgodzik) [inline .scalafix.conf:4]: What does this rule do?

**Review by tgodzik — COMMENTED** [review]
Looks sensible to me, did you try running scalafix? I remember I had some issues with it, but I tried fixing some dependencies causing those issues, so might be ok now.

File: .scalafix.conf (hunk 1 above)
Comment (kpodsiad) [reply .scalafix.conf:4]: it bans 
``` scala
for {
  ... <- ...
  val x = ??? // <- this
} yield ...
```

Comment (tgodzik) [reply .scalafix.conf:4]: makes sense!

**Comment by kpodsiad** [comment]
I think there's no point in adding scalafix to the CI for now - first we should handle initial sbt load overhead.

**Comment by tgodzik** [comment]
> I think there's no point in adding scalafix to the CI for now - first we should handle initial sbt load overhead.

We can just add it to a separate job, no?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -6,6 +6,9 @@ object Dependencies {
   val Scala212Version = "2.12.15"
   val Scala213Version = "2.13.8"
 
+  val Scala211BinaryVersion = "2.11"
```
Comment (tgodzik) [inline project/Dependencies.scala:9]: Is this needed? I can't find it being used.

**Comment by kpodsiad** [comment]
I'll do `rewrite generation of the test projects to not happen on sbt load` task from Bloop next steps and then I'll add a CI job.

**Review by tgodzik — APPROVED** [review]
LGTM! Thanks for doing this!

**Pull request merged**